### PR TITLE
Add PHP 8.3 and bump actions/checkout

### PR DESCRIPTION
**Pull request by williamdes**

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 name: JSON Serializer CI
 
+permissions:
```
Comment (jrbasso) [inline .github/workflows/php.yml:3]: @williamdes Why did you have to add this?

Comment (williamdes) [reply .github/workflows/php.yml:3]: It's not required, but if missing the permissions given to the CI token are wider and more insecure if someone manages to steal the token

Comment (williamdes) [reply .github/workflows/php.yml:3]: https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

**Comment by jrbasso** [comment]
Thank you for your contribution!

**Pull request merged**